src/coredump/coredump.c

```diff
@@ -819,6 +825,26 @@ static int submit_coredump(
                 (void) coredump_vacuum(coredump_node_fd >= 0 ? coredump_node_fd : coredump_fd, arg_keep_free, arg_max_use);
         }
 
+        if (mount_tree_fd >= 0 && arg_access_container) {
+                r = unshare(CLONE_NEWNS);
+                if (r < 0) {
+                        log_warning_errno(errno, "Failed to unshare mount namespace, not using passed mount tree: %m");
+                        mount_tree_fd = safe_close(mount_tree_fd);
+                } else {
+                        r = mount(NULL, "/", NULL, MS_REC|MS_PRIVATE, NULL);
```

Comment:
just to be safe against seccomp/lsm I'd actually move all these 4 calls into a static helper, and if the helper fails then log and continue